docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
 
-## Enabling search of `.flux.yaml` files
+This feature lets you generate Kubernetes manifests with a program,
+instead of having to include them in your git repo as YAML files. For
+example, you can use `kustomize` to patch a common set of resources to
+suit a particular environment.
 
-To enable it supply the command-line flag `--manifest-generation=true` to `fluxd`.
+## Setting manifest generation up
 
-## Goal
+Manifest generation is controlled by the flags given to `fluxd`, and
+`.flux.yaml` files in your git repo.
 
-It is a common pattern to run very similar resources in separate clusters. There are various scenarios in which this is
-required, the two main ones being:
+To enable it, pass the command-line flag `--manifest-generation=true`
+to `fluxd`.
 
-* Having a staging/canary cluster and production cluster. The resources from the staging cluster are regularly promoted
-  to the production cluster. In addition, there are long-term differences between the cluster resources (e.g. different
-  security keys, different database endpoints etc ...).
-* Federation. Different clusters run in separate regions, usually with very similar resources but different
-  configurations.
+For each path supplied with the flag `--git-path` (or `.` if none are
+supplied),
 
-The main goal of `.flux.yaml` configuration files is to help deploying similar resources/clusters:
+ - if a `.flux.yaml` file is found, it is used **instead** of looking
+   for YAML files;
 
-* with minimal replication of resource definitions
-* while keeping Flux neutral about the factorization technology used
+ - if no `.flux.yaml` file is found, the usual behaviour of looking
+   for YAML files is used.
 
-## File-access behaviour in Flux
+The manifests from all the `--git-path` values -- read from YAML files
+or generated -- are combined before applying to the cluster. If
+duplicates are detected, an error is logged and the sync is abandoned.
 
-Flux performs two types of actions on raw manifest files from the Git repository:
+A `.flux.yaml` file can be _at_ the path mentioned, or in a directory
+_above_ it in the git repo. Here are some examples:
 
-1. Read manifest files when performing a sync operation (i.e making sure that the status of the cluster reflects what's
-   in the manifest files, adjusting it if necessary)
-2. Update the manifest files of [workload](fluxctl.html#workloads).
-   Specifically, flux can update:
-    * container images, when releasing a new image version. A release can happen manually or automatically, when a new
-      container image is pushed to a repository.
-    * annotations, which establish the release policy of a workload (e.g. whether it should be automatically released,
-      whether it should be locked from releasing, what image tags should be considered for automated releases …)
+```sh
+# show normal files _and_ .flux.yaml
+flux-kustomization-example$ tree -a -P .flux.yaml -P '*' -I .git
```

Comment:
Is the _sophisticated_ tree command and `flux-kustomization-example` name necessary? I think it doesn't add much value as is and can add confusion.

I think we should either be explicit about using a real-world example (even adding a prior `git clone` command on https://github.com/weaveworks/flux-kustomize-example ) or just show the tree.

Reply:
I would go with a `git clone && cd flux-kustomization-example` command and remove `flux-kustomization-example$`.

Reply:
BTW, I should move the repo to the `fluxcd` org.

@squaremo Do you want me to do it before merging this PR?

Reply:
GitHub deals with redirects, I would just move it now.

Reply:
I wanted to show a directory structure, but it doesn't really matter how I get it, so I'll just remove the command from the examples.